File: study-kotlin/java-to-kotlin/kotlin/src/test/kotlin/lec10_inheritance/javacode/Penguin.java
package lec10_inheritance.javacode;

public class Penguin
	extends Animal
	implements Flyable, Swimable {

	private final int wingCount;

	public Penguin(final String species, final int legCount) {
		super(species, legCount);
		this.wingCount = 2;
	}

	@Override
	public void move() {
		System.out.println("펭귄 펭귄");
	}

	@Override
	public int getLegCount() {
		return super.getLegCount() + this.wingCount;
	}

	@Override
	public void act() {
		Swimable.super.act();
		Flyable.super.act();
	}

	@Override
	public void fly() {
		System.out.println("fly me to the moon");
	}
}

File: study-kotlin/java-to-kotlin/kotlin/src/test/kotlin/lec10_inheritance/code/Penguin.kt
package lec10_inheritance.code

class Penguin(
    species: String
) : Animal(species, 2), Swimable, Flyable {
    private val wingCount = 2

    override fun move() {
        println("펭귄 펭귄")
    }

    override val legCount: Int
        get() = super.legCount + this.wingCount

    override fun act() {
        super<Swimable>.act()
        super<Flyable>.act()
    }

    override fun fly() {
        println("fly me to the moon")
    }

    override val swimAbility: Int
        get() = 3
}